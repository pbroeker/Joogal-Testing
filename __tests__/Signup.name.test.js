import React from 'react';
import Name from '../Signup/Signup.Name';
import { fireEvent, render } from '@testing-library/react-native';
import { Alert } from 'react-native';

jest.spyOn(Alert, 'alert').mockImplementation(() => {});

describe('Signup Name', () => {
  const setSignup = jest.fn();
  const navigation = {navigate : jest.fn()};
  let renderedComponent;

  beforeEach(() => {
    Alert.alert.mockReset();
    renderedComponent = render( <Name navigation={navigation} setSignup={setSignup} /> );
  })

  test('snapshot', () => {
    const { toJSON } = renderedComponent;
    expect(toJSON()).toMatchSnapshot();
  })

  test('rendered correctly', () => {
    const { getByText } = renderedComponent;
    expect(getByText('What\'s your name?')).toBeTruthy();
  })

  test('calls signup correctly', () => {
    const { getByText, getByPlaceholderText } = renderedComponent;
    fireEvent(getByPlaceholderText('First name*'), 'onChangeText', 'testuser');
    fireEvent(getByText('➤'), 'onPress');
    expect(setSignup).toHaveBeenCalledTimes(1);
  })

  test('calls alert if user doesn\' give any firstname', () => {
    const { getByText } = renderedComponent;
    fireEvent(getByText('➤'), 'onPress');
    expect(Alert.alert).toHaveBeenCalledWith('Please enter at least your first name');
  })

})