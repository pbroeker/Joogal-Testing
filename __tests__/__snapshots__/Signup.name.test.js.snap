// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Name snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1d1e20",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "marginTop": "20%",
        "position": "absolute",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "whitesmoke",
          "fontFamily": "PoiretOne_400Regular",
          "fontSize": 30,
          "marginBottom": 30,
          "marginLeft": "10%",
          "width": "53%",
        }
      }
    >
      What's your name?
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": "45%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCompleteType="name"
      autoFocus={true}
      onChangeText={[Function]}
      placeholder="First name*"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "grey",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "whitesmoke",
          "marginVertical": "1%",
          "padding": 14,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Last name"
      placeholderTextColor="grey"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "grey",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": "whitesmoke",
          "marginVertical": "1%",
          "padding": 14,
          "width": "80%",
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "marginTop": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "teal",
          "borderRadius": 50,
          "marginRight": 30,
          "opacity": 1,
          "padding": 10,
          "width": 56,
        }
      }
      testID="nameContinueButton"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 30,
            "fontWeight": "600",
          }
        }
      >
        âž¤
      </Text>
    </View>
  </View>
</View>
`;
